### PR TITLE
Add additional log + retry in case of empty config file

### PR DESCRIPTION
**Pull request by atobiszei**
Add retry in case of spurious race in reading file vs reading modification time.
This should give us more information in case of intermittent failure while most probably fixing with workaround as well.

JIRA:CVS-61368

### CODE REVIEW

File: src/modelmanager.cpp [hunk 1]
```diff
@@ -530,28 +530,72 @@ Status ModelManager::tryReloadGatedModelConfigs(std::vector<ModelConfig>& gatedM
...
+           << " time modification [s]: " << statTime.st_ctim.tv_sec
+           << " [ns]: " << statTime.st_ctim.tv_nsec << std::endl;
+        std::ifstream file(filename);
+        std::string some;
+        while (file) {
+            file >> some;
+            ss << some << std::endl;
+        }
+    }
+    void log(const std::string com = "") {
```
Comment (dkalinowski) [inline src/modelmanager.cpp:559]: The parameter is not used

File: src/modelmanager.cpp [hunk 2]
```diff
@@ -530,28 +530,72 @@ Status ModelManager::tryReloadGatedModelConfigs(std::vector<ModelConfig>& gatedM
...
+        SPDLOG_LOGGER_DEBUG(modelmanager_logger, ss.str());
+    }
+    ~LoudFileInfoReporter() {
```
Comment (dkalinowski) [inline src/modelmanager.cpp:562]: Why do we need this destructor implementation?

File: src/modelmanager.cpp [hunk 3]
```diff
@@ -530,28 +530,72 @@ Status ModelManager::tryReloadGatedModelConfigs(std::vector<ModelConfig>& gatedM
...
+    LoudFileInfoReporter(const std::string& name) :
+        filename(name) {
+        if (filename == "") {
+            return;
+        }
+        struct stat statTime;
+
+        if (stat(filename.c_str(), &statTime) != 0) {
+            SPDLOG_ERROR("Failed to debug-read fileconfig");
+            return;
```
Comment (dkalinowski) [inline src/modelmanager.cpp:547]: What's the point of checking stat before opening the file?

File: src/modelmanager.cpp [hunk 4]
```diff
@@ -530,28 +530,72 @@ Status ModelManager::tryReloadGatedModelConfigs(std::vector<ModelConfig>& gatedM
...
+        rapidjson::IStreamWrapper isw(ifs);
+        rapidjson::ParseResult parseResult = configJson.ParseStream(isw);
+        if (!parseResult) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Configuration file is not a valid JSON file. Error: {}",
+                rapidjson::GetParseError_En(parseResult.Code()));
+            intermediateStatus = StatusCode::JSON_INVALID;
+            loud.log(std::to_string(counter));
+            continue;
+        }
+    } while (++counter < 2 && !intermediateStatus.ok());
```
Comment (dkalinowski) [inline src/modelmanager.cpp:596]: Create const variable holding the `2` parameter with descriptive name (for example CONFIG_JSON_READ_RETRY_COUNT)

File: src/modelmanager.cpp (hunk 3 above)
Comment (atobiszei) [reply src/modelmanager.cpp:543]: To leave time stamp as debug info

**Review by dkalinowski — APPROVED** [review]

**Review by mzegla — APPROVED** [review]

**Pull request merged**